### PR TITLE
Makefile:  Use `cabal doctest`

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

**Review by geekosaur — APPROVED** [review]

**Comment by geekosaur** [comment]
Looks like `cabal-doctest` has a problem with `--allow-newer`? https://github.com/haskell/cabal/actions/runs/10119868982/job/27988903129?pr=10231#step:9:12

**Comment by sol** [comment]
> Looks like `cabal-doctest` has a problem with `--allow-newer`?

Actually, all options with optional arguments are affected.

This will be fixed by https://github.com/sol/doctest/pull/445.

File: Makefile [hunk 1]
```diff
@@ -107,11 +104,11 @@ ghcid-cli :
...
-	$(DOCTEST) Cabal-syntax
-	$(DOCTEST) Cabal-described
-	$(DOCTEST) --build-depends=QuickCheck Cabal
-	$(DOCTEST) cabal-install-solver
-	$(DOCTEST) cabal-install
+	cd Cabal-syntax && $(DOCTEST)
+	cd Cabal-described && $(DOCTEST)
+	cd Cabal && $(DOCTEST)
+	cd cabal-install-solver && $(DOCTEST)
+	cd cabal-install && $(DOCTEST)
```
Comment (sol) [inline Makefile:111]: @geekosaur changing the working directory is necessary as per https://github.com/haskell/cabal/issues/9115#issuecomment-2254293615.

Other than that, this is good to go.

**Review by ulysses4ever — APPROVED** [review]
I don't love those `cd`'s but I think we should be conducive to contributions that exercise cabal latest features (e.g. external commands). A big +1 from me.

**Comment by sol** [comment]
> I don't love those `cd`'s

I definitely agree.  I think #9115 is the way to fix that.

**Pull request merged**